examples/image_reconstruction.ipynb

```diff
@@ -0,0 +1,705 @@
+    "print ('The running processor is...', device)\n",
+    "set_seeds(42)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "M56Tvbbvw_P1"
+   },
+   "source": [
+    "The MNIST dataset is directly donlowded from torchvision.datasets. The training dataset contains 60,000 images of digits at the size of 28x28, However we seperate 20% of the full train dataset as validation dataset. On the other hand by passing \"train\" argument as False, the test dataset, containing 10,000 images is downloaded and saved in the dataset directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "QVJ3WG_hw_P1"
+   },
+   "outputs": [],
+   "source": [
+    "# This section produces the original MNIST dataset, which is provided for classification applications. ---> (image, lable)\n",
+    "# In the next section we should redefine our datasets to make it proper for the reconstruction application. ---> (image, image)\n",
+    "\n",
+    "\n",
+    "if not os.path.exists('dataset'): os.mkdir('dataset')   # making a directory to download and save the datasets\n",
+    "\n",
+    "full_train_dataset = datasets.MNIST('./dataset/', train=True, download=True, transform=tfms.ToTensor())\n",
+    "test_dataset = datasets.MNIST('./dataset/', train=False, download=True, transform=tfms.ToTensor())\n",
+    "\n",
+    "# Selecting and seperating rate% (for example 20%) of the full train dataset for as validation dataset\n",
+    "indices = list(np.arange(len(full_train_dataset)))\n",
+    "np.random.shuffle(indices)\n",
+    "trn_indices = indices[math.floor(len(full_train_dataset)*val_split_percent):]\n",
+    "val_indices = indices[:math.floor(len(full_train_dataset)*val_split_percent)]\n",
+    "\n",
+    "\n",
+    "trn_dataset = Subset(full_train_dataset, trn_indices)\n",
+    "val_dataset = Subset(full_train_dataset, val_indices)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "lXTbdOE1w_P1"
+   },
+   "source": [
+    "The original downloaded MNIST dataset, is provided fin classification application format, which means each sample is compsed of an image and the lable (eual to the number which that images shows). However, for image classification the dataset should contain an input image and a target image, which simply are the same. Hense, by the code below, we define a new dataset, which provides an image as an input and sets that image as its target as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "vWtcE1zIw_P2"
+   },
+   "outputs": [],
+   "source": [
+    "#This function changes the original MNIST dataset (originally proper for classification, as it contains labels) \n",
+    "#to a dataset proper for image reconstruction.\n",
+    "#which changes the pair of (image,label) to a (image, image).\n",
+    "\n",
+    "class mydataset(Dataset):       \n",
```

Comment:
The name of the class is not very nice. I would use a name like `ImageReconstructionDataset` instead.